protocol/test/forge/v2/safeGuards/ThresholdSafeGuard.t.sol

```diff
@@ -498,8 +500,178 @@ contract ThresholdSafeGuardTest is ThresholdSafeGuardTestBase {
 
             changePrank(address(safe));
             vm.expectRevert("!Dynamic Signature Threshold");
-            doCheck(fnCall, safeOwners[0], signature);
+            doCheck(fnCall, 0, safeOwners[0], signature);
         }
     }
 
 }
+
+contract ThresholdSafeGuardExecuteTest is ThresholdSafeGuardTestBase {
+
+    function _setup() internal returns (uint256 pk2, uint256 pk3) {
+        // Set the guard
+        vm.startPrank(address(safe));
+        safe.setGuard(address(guard));
+
+        // Fund the safe some ETH
+        vm.deal(address(safe), 200e18);
+
+        // Add a couple more executors to the safe which we can sign on behalf of.
+        (pk2, pk3) = addSafeOwners();
+    }
+
+    function test_checkSafeExecute_withOverrideTresholdSet_ethTransfer() public {
+        (uint256 pk2, uint256 pk3) = _setup();
+
+        changePrank(executor);
+        bytes4 fnSelector = bytes4(0);
+        guard.setFunctionThreshold(address(mock), fnSelector, 3);
+
+        // An empty function call (eg ETH transfer)
+        bytes memory fnCall = bytes("");
+
+        bytes memory txData = safe.encodeTransactionData({
+            to: address(mock), 
+            value: 100e18,  
+            data: fnCall,
+            operation: Enum.Operation.Call,
+            safeTxGas: 0,
+            baseGas: 0,
+            gasPrice: 0,
+            gasToken: address(0),
+            refundReceiver: payable(0),
+            _nonce: safe.nonce()
+        });
+        bytes32 txDataHash = keccak256(txData);
+
+        vm.startPrank(safeOwners[0]);
```

Comment:
nit: for consistency, I would use `changePrank` here too

Reply:
It's a little annoying in foundry -- `changePrank()` can't be used first as internally just calls:
```
        vm.stopPrank();
        vm.startPrank(msgSender);
```

And `vm.stopPrank()` will fail if an existing prank isn't running yet.